scipy/stats/_binned_statistic.py

```diff
@@ -650,6 +633,41 @@ def binned_statistic_dd(sample, values, statistic='mean',
     return BinnedStatisticddResult(result, edges, binnumbers)
 
 
+def _calc_binned_statistic(Vdim, binnumbers, result, values, stat_func):
+    unique_bin_numbers = np.unique(binnumbers)
+    for vv in builtins.range(Vdim):
+        bin_map = _create_bin_map(binnumbers, unique_bin_numbers, values, vv)
+        for i in unique_bin_numbers:
+            result[vv, i] = stat_func(bin_map[i])
+
+
+def _calc_binned_std(Vdim, binnumbers, result, values):
+    result.fill(0)
+    unique_bin_numbers = np.unique(binnumbers)
+    for vv in builtins.range(Vdim):
+        bin_map = _create_bin_map(binnumbers, unique_bin_numbers, values,
+                                  vv)
+        for i in unique_bin_numbers:
+            # NOTE: take std dev by bin, np.std() is 2-pass and stable
+            binned_data = bin_map[i]
+            # calc std only when binned data is 2 or more for speed up.
+            if len(binned_data) >= 2:
+                result[vv, i] = np.std(binned_data)
+
+
+def _create_bin_map(binnumbers, unique_bin_numbers, values, vv):
+    """ Create bin map
+    key: bin number
+    value: list of binned data
```

Comment:
Ah this part helps clarify, the key is the bin number (unique) and the value is a list (or array-like) of binned data. Maybe the reason why I'm confused is because of conflating the word `map` with `hashmap`, can we maybe call the function `_create_bin_hashmap`? 

Not sure if this is just me or if it is (or will be) genuinely confusing to others too.

Reply:
I agree. I changed it to `_create_bin_hashmap`.